src/audio/buffer.c

```diff
@@ -27,6 +27,7 @@ uintptr_t buffer_uuid_ptr = SOF_UUID(buffer_uuid);
 struct comp_buffer *buffer_alloc(uint32_t size, uint32_t caps, uint32_t align)
 {
 	struct comp_buffer *buffer;
+	uint32_t mem_caps = caps &= ~0xFFFF0000;
```

Comment:
Why not 
```suggestion
	uint32_t mem_caps = caps &= 0xFFFF;
```

Reply:
@xiulipan hmm that doesn't work for me.... I guess the hex constants define the number of bits also. So your suggestion will be become 16 bit unsigned, then promoted to 32 bit -> so 16 msb's are still zero.

Reply:
@xiuli sorry didn't notice that you removed the ~. 

anyway what ever is more clear for people to understand what I'm trying to do here...

Reply:
@juimonen Try 0xFFFFL to make compiler take this value as 0x0000FFFF